File: GO/Colleen.go
package main

import "fmt"

// commentaire en dehors

func print() {
    str := "package main%[1]c%[1]cimport %[2]cfmt%[2]c%[1]c%[1]c// commentaire en dehors%[1]c%[1]cfunc print() {%[1]c    str := %[2]c%[3]s%[2]c%[1]c    fmt.Printf(string(str), 10, 34, str)%[1]c}%[1]c%[1]cfunc main() {%[1]c    // commentaire dedans%[1]c    print()%[1]c}"
    fmt.Printf(string(str), 10, 34, str)
}

func main() {
    // commentaire dedans
    print()
}

File: C/Colleen.c
#include <stdio.h>

/*
    Un commentaire en dehors
*/

void print () {
    char *str = "#include <stdio.h>%1$c%1$c/*%1$c    Un commentaire en dehors%1$c*/%1$c%1$cvoid print () {%1$c    char *str = %3$c%2$s%3$c;%1$c    printf(str, 10, str, 34);%1$c    return;%1$c}%1$c%1$cint main() {%1$c    /*%1$c        Un commentaire dedans%1$c    */%1$c    print();%1$c    return 0;%1$c}%1$c";
    printf(str, 10, str, 34);
    return;
}

int main() {
    /*
        Un commentaire dedans
    */
    print();
    return 0;
}
